sale_purchase_requisition/models/sale_order.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    purchase_agent = fields.Many2one(
+        comodel_name="res.users", string="Comercial compra"
+    )
+
+    purchase_requisition_ids = fields.One2many(
+        comodel_name="purchase.requisition",
+        inverse_name="sale_order_id",
+        string="Acuerdo de compra",
+    )
+
+    def action_view_sale_requisition(self):
+        return {
+            "name": _("Purchase Requisition"),
+            "type": "ir.actions.act_window",
+            "res_model": "purchase.requisition",
+            "view_mode": "form",
+            "res_id": self.purchase_requisition_ids.id,
+        }
+
+    def _create_lines_preq(self):
```

Comment:
Better rename to `_prepare_purchase_requisition_lines`.
Odoo standard of naming functions which returns prepared data for creations are called starting with `_prepare`.